### DISCUSSION TITLE
Provisional documents

### DISCUSSION THREAD

**Discussion by Keshaun-Plus** [keylime/keylime#973] [Q&A]
When I used Keylime for remote attestation, I ran into an error. Here's why:
1. When I look at the IMA file, I find some file-paths in IMA file that don't actually exist（Provisional documents）;
2. Therefore, I do not include those files in my whitelist;
3. However, when remote attestation, these file paths that do not actually exist will be compared with the whitelist, and then errors will occur;
It might be possible to add these file-paths to exclude.txt, but how can i ensure that system do not generate new provisional files(file-path in IMA file that don't actually exist) during the remote attestation?
Pliease help me

**Comment by Keshaun-Plus** [comment]
and have you encountered this problem（Provisional documents？）？

**Reply by THS-on** [reply]
Can you post our IMA configuration and an entry of a file that you cannot find the hash for?

What do you exactly mean with "Provisional documents"? Do you mean logs, configuration files or some programs that are only included in the initrd (initramfs)?

**Comment by mpeters** [comment]
To expand on what @THS-on said, the solution is probably to be found in your IMA configuration. If there are things you want to exclude from measurement that will happen on the IMA side, otherwise it will be in your IMA measurement list and thus in your TPM. You could make some broad rules in an excludelist file, but it's easier to just exclude it on the IMA side to begin with.
